synapse/lib/schemas.py

```diff
@@ -668,9 +668,22 @@
             'pattern': s_version.semverstr,
         },
         'build': {
-            'type' 'object'
+            'type': 'object',
             'properties': {
                 'time': {'type': 'number'},
+                'synapse:verstring': {
+                    'type': 'string',
+                    'pattern': s_version.semverstr
+                },
+                'synapse:version': {
+                    'type': 'array',
+                    'items': {'type': 'number'}
+                },
+                'synapse:commit': {
+                    'type': 'string',
+                    # Note: This pattern allows empty string for dev environments
```

Comment:
Wouldn't we just not set it?

Reply:
The value of `synapse.lib.version.commit` is an empty string; we punch that value in when we build containers or when doing wheel / sdist builds for redistribution.

Reply:
So should I omit the key if the value is empty?

Reply:
i'm on the fence - I could go either way really. either an empty string OR a git commit hash; but not both.

Reply:
I left this the way it is. The current pattern allows an empty string or a hex hash of any length. (I was thinking about future proofing when git decided to update their hashes to SHA512 or something).